modules/nextflow/src/test/groovy/nextflow/config/ConfigBuilderTest.groovy

```diff
@@ -2229,5 +2236,37 @@ class ConfigBuilderTest extends Specification {
         folder?.deleteDir()
     }
 
+    @Unroll
+    def 'should create builder for given engine' () {
+        given:
+        def IMAGE = 'foo:latest'
+
+        when:
+        def builder = ContainerBuilder.create(ENGINE,IMAGE)
+        then:
+        builder.class == CLAZZ
+        builder.getImage() == IMAGE
+
+        where:
+        ENGINE              | CLAZZ
+        'docker'            | DockerBuilder
+        'podman'            | PodmanBuilder
+        'singularity'       | SingularityBuilder
+        'shifter'           | ShifterBuilder
+        'charliecloud'      | CharliecloudBuilder
+        'udocker'           | UdockerBuilder
+
+    }
```

Comment:
Shouldn't this all be in the `ContainerBuilderTest` ?

Reply:
Yes, you are right